### PR TITLE
Initial build for 0.3.14

### PR DESCRIPTION
**Pull request by JeanChristopheMorinPerso**
tenseal 0.3.14 :snowflake:

**Destination channel:** Snowflake

### Links

- [PKG-3760](https://anaconda.atlassian.net/browse/PKG-3760)
- [Upstream repository](https://github.com/OpenMined/TenSEAL/tree/v0.3.14)
- [Upstream changelog/diff]()
- Relevant dependency PRs:
  - https://github.com/AnacondaRecipes/xtl-feedstock/pull/1
  - https://github.com/AnacondaRecipes/ms-gsl-feedstock/pull/1
  - https://github.com/AnacondaRecipes/xsimd-feedstock/pull/8
  - https://github.com/AnacondaRecipes/xtensor-feedstock/pull/1
  - https://github.com/AnacondaRecipes/seal-feedstock/pull/1

### Explanation of changes:

New feedstock. It doesn't exist on conda-forge.

Note that I had to unvendor all dependencies via a patch.

[PKG-3760]: https://anaconda.atlassian.net/browse/PKG-3760?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by SC426 — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - libprotobuf  # pin through run_exports
+    - numpy
+
+test:
+  source_files:
+    - tests/python
+    - tests/__init__.py
+  imports:
+    - tenseal
+    - tenseal.sealapi
```
Comment (cbouss) [inline recipe/meta.yaml:53]:
```suggestion
    - tenseal.sealapi
    - tenseal.tensors
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  run:
+    - python
+    - zlib  # pin through run_exports
+    - libprotobuf  # pin through run_exports
+    - numpy
+
+test:
+  source_files:
+    - tests/python
+    - tests/__init__.py
```
Comment (cbouss) [inline recipe/meta.yaml:49]:
```suggestion
    - tests/__init__.py
    - pytest.ini
```
(Required for pytest to recognize the slow marker.

**Review by cbouss — APPROVED** [review]
LGTM, great work on the cmake patch!
I added minor suggestions on the tests.

File: recipe/meta.yaml (hunk 2 above)
Comment (JeanChristopheMorinPerso) [reply recipe/meta.yaml:49]: It's not necessary and it just removes the pytest warning. But I can add it to remove the warning.

**Pull request merged**